python/cuda_parallel/cuda/parallel/experimental/_iterators.py

```diff
@@ -96,18 +96,18 @@ def __init__(self, ptr, ntype):
         data_as_ntype_pp = numba.types.CPointer(numba.types.CPointer(ntype))
         self.ntype = ntype
         self.prefix = "pointer_" + ntype.name
-        self.ltoirs = [
+        self.ltoirs, _ = zip(*[
```

Comment:
Could we just change `_ncc()` instead, to strip the `resty` right there? (The `zip` trick isn't exactly intuitive, at least to me.)

Reply:
I agree. I changed this pattern by making it so that `_ncc` returns a `namedtuple`. And we just do:

```python
ltoirs = [
   _ncc(...).ltoir,
   _ncc(...).ltoir,
  ...
]
```

Which I think is a bit more readable compared to the alternative:

```python
ltoirs = [
   _ncc(...)[0],
   _ncc(...)[0],
  ...
]
```